core/domain/skill_domain.py

```diff
@@ -722,24 +722,20 @@ def update_explanation(self, explanation):
         Args:
             explanation: SubtitledHtml. The new explanation of the skill.
         """
-        self.skill_contents.explanation = (
-            state_domain.SubtitledHtml.from_dict(explanation))
+        self.skill_contents.explanation = (explanation)
 
     def update_worked_examples(self, worked_examples):
         """Updates the worked examples list of the skill.
 
         Args:
-            worked_examples: list(dict). The new worked examples of the skill.
+            worked_examples: list(SubtitledHtml). The new worked examples
+            of the skill.
         """
         old_content_ids = [worked_example.content_id for worked_example in (
-            self.skill_contents.worked_examples)]
-
-        self.skill_contents.worked_examples = [
```

Comment:
I think you have missed updating the `self.skill_contents.worked_examples` with the worked_examples.

Reply:
Thanks @DubeySandeep I will fix the requested changes.

Reply:
Done